File: src/backend/studio-reservation/src/main/java/org/taonaw/studio_reservation/domain/model/studio/StudioId.java
package org.taonaw.studio_reservation.domain.model.studio;

import de.huxhorn.sulky.ulid.ULID;
import lombok.EqualsAndHashCode;
import lombok.Getter;
import lombok.NonNull;
import org.taonaw.studio_reservation.domain.shared.Assertion;

@Getter
@EqualsAndHashCode
public class StudioId {
    private final String value;

    public StudioId(String value) {
        Assertion.required(value);
        this.value = value;
    }

    public static StudioId newId() {
        return new StudioId(new ULID().nextULID());
    }
}

File: src/backend/facility/src/main/kotlin/org/taonaw/facility/domain/model/studio/StudioId.kt
package org.taonaw.facility.domain.model.studio

import de.huxhorn.sulky.ulid.ULID

data class StudioId(val value: String) {
    companion object {
        fun newId(): StudioId {
            return StudioId(ULID().nextULID())
        }
    }
}